### PR TITLE
gh-124083: Skip test_strsignal on NetBSD due to TypeError

### PR DESCRIPTION
**Pull request by furkanonder**
* Issue: gh-124083

### CODE REVIEW

File: Lib/test/test_signal.py [hunk 1]
```diff
@@ -124,6 +124,8 @@ def __repr__(self):
         self.assertEqual(0, argument.repr_count)
 
     def test_strsignal(self):
+        if sys.platform.startswith("netbsd"):
```
Comment (rruuaanng) [inline Lib/test/test_signal.py]: Maybe you should add a NEWS to account for your modifications, Because it may confuse unfamiliar people, That's i think.

Comment (Wulian233) [reply Lib/test/test_signal.py]: tests usually don't need NEWS, and in part, this is invisible to users

**Comment by Zheaoli** [comment]
I think we need to link this PR to the NetBSD community. We don't know whether this issue is caused by the NetBSD in a specific version or this is a wide exist bug.

I think we need to hold on this PR before we confirm the bug scope

**Review by picnixz — COMMENTED** [review]
If `strsignal` is not supported on this platform, then the `configure` check must have detected it. However, it appears that your C program works correctly (namely that `strsignal` is present), so `HAVE_STRSIGNAL` should be defined. This also means that the following somehow returns None.

```C
    errno = 0;
    res = strsignal(signalnum);

    if (errno || res == NULL || strstr(res, "Unknown signal") != NULL)
        Py_RETURN_NONE;
```

Either `errno` is actually set differently or `res` is NULL. But this seems quite weird that `res` is NULL since the C program works. So `errno` is somewhat modified. 

If `HAVE_STRSIGNAL` is not specified, then the `switch` we are using is wrong.  We should also check that the constants being check are actually the correct ones. I don't have access to a NetBSD machine but something more delicate could be hidden.

**Comment by Zheaoli** [comment]
> If `HAVE_STRSIGNAL` is not specified, then the `switch` we are using is wrong. We should also check that the constants being check are actually the correct ones. I don't have access to a NetBSD machine but something more delicate could be hidden.

I'm working on the root cause. This is a little bit wired

File: Lib/test/test_signal.py [hunk 2]
```diff
@@ -124,6 +124,8 @@ def __repr__(self):
         self.assertEqual(0, argument.repr_count)
 
     def test_strsignal(self):
```
Comment (vstinner) [inline Lib/test/test_signal.py:128]: Can you use a decorator on the method instead? Something like:

```
@unittest.skipIf(sys.platform.startswith("netbsd"), "gh-124083: strsignal is not supported on NetBSD")
```

**Comment by vstinner** [comment]
@Zheaoli: There are more details in the issue: https://github.com/python/cpython/issues/124083#issuecomment-2350751764 `signal.strsignal()` returns None for most signals.

**Review by vstinner — APPROVED** [review]
LGTM

**Comment by vstinner** [comment]
Merged, thanks for the fix @furkanonder.

**Comment by furkanonder** [comment]
> Merged, thanks for the fix @furkanonder.

You are welcome!

**Comment by picnixz** [comment]
I'm still convinced that the fact it returns None is something that should be investigated. But I can live with this fix for now

**Comment by vstinner** [comment]
> Then why does it work with the C example on the issue?

If fails with errno ``2``.

**Comment by picnixz** [comment]
Oh! I thought the result would be NULL in this case! My bad. Ok so nothing to investigate. Sorry for the noise :')

**Comment by vstinner** [comment]
No need to be sorry, strsignal() on NetBSD has a strange behavior (return a string **and** report an error). IMO it's easier to skip the test, there is no need to spend too much time on it until NetBSD is fixed.

**Pull request merged**